pep-0712.rst

```diff
@@ -208,56 +225,66 @@ Example
 
+Backward Compatibility
+======================
+
+These changes don't introduce any compatibility problems since they
+only introduce new features.
+
+Security Implications
+======================
+
+There are no direct security concerns with these changes.
+
+How to Teach This
+=================
+
+Documentation and examples explaining the new parameter and behavior will be
+added to the relevant sections of the docs site (primarily on
+:mod:`dataclasses`) and linked from the *What's New* document.
+
 
 Reference Implementation
 ========================
 
-The `attrs <#attrs-converters>`_ library already includes a ``converter``
+The attrs library `already includes <attrs-converters_>`__ a ``converter``
 parameter containing converter semantics.
 
-CPython support can be seen on a branch: `GitHub <#attrs-converters>`_.
+CPython support is implemented on `a branch in the author's fork <cpython-branch_>`__.
 
 Rejected Ideas
 ==============
 
-Just adding "converter" to ``dataclass_transform``'s ``field_specifiers``
--------------------------------------------------------------------------
+Just adding "converter" to :func:`~py3.11:~typing.dataclass_transform`'s ``field_specifiers``
+---------------------------------------------------------------------------------------------
 
-The idea of isolating this addition to ``dataclass_transform`` was briefly
-discussed in `Typing-sig <#only-dataclass-transform>`_ where it was suggested
-to open this to ``dataclasses``.
+The idea of isolating this addition to
+:func:`~py3.11:~typing.dataclass_transform` was briefly
+`discussed on Typing-SIG <only-dataclass-transform_>`__ where it was suggested
+to broaden this to :mod:`dataclasses` more generally.
 
-Additionally, adding this to ``dataclasses`` ensures anyone can reap the
+Additionally, adding this to :mod:`dataclasses` ensures anyone can reap the
 benefits without requiring additional libraries.
 
 Automatic conversion using the field's type
 -------------------------------------------
 
 One idea could be to allow the type of the field specified (e.g. ``str`` or
 ``int``) to be used as a converter for each argument provided.
-`Pydantic's data conversion <#pydantic-data-conversion>`_ has semantics which
+`Pydantic's data conversion <pydantic-data-conversion_>`__ has semantics which
 appear to be similar to this approach.
 
 This works well for fairly simple types, but leads to ambiguity in expected
 behavior for complex types such as generics. E.g. For ``tuple[int]`` it is
 ambiguous if the converter is supposed to simply convert an iterable to a tuple,
 or if it is additionally supposed to convert each element type to ``int``.
 
-Converting the default values
------------------------------
-
-Having the synthesized ``__init__`` also convert the default values (such as
-``default`` or the return type of ``default_factory``) when the would make the
-expected type of these parameters complex for type-checkers, and does not add
-significant value.
```

Comment:
You might want to add the inverse of this, not converting the default values, as a rejected idea with an appropriate explanation particularly given you proposed it initially.

Reply:
That is a fantastic idea!